### PR TITLE
Fix volunteer map load time

### PR DESCRIPTION
**Pull request by hacodeorg**
# Description
[Local volunteer map](https://code.org/volunteer/local) has a performance issue on **first load.** The page currently takes about 15-30s to load (without filter) because it tries to display 5000 volunteers at once. This will significantly impact users experience when they want helps for HoC.

After investigation, the main reason for the slowness is to render 5000 pins on Google Maps. 
The fix is to reduce the number of pins on **first load** by reducing search distance and volunteer count. Specifically, I set the starting search location to Seattle, with 20 mile radius and max 50 volunteers. 

**Screenshots**
First load:
Screen Shot 2019-12-03 at 4 20 25 PM

Searches specific location with default distance (20 miles) and volunteer count (50):
Screen Shot 2019-12-03 at 4 18 21 PM

Examples of further filtering:
Screen Shot 2019-12-03 at 4 21 31 PM

## Links
- [jira](https://codedotorg.atlassian.net/browse/PLC-592)

## Testing story
- Tested on local machines using fake 10k volunteers

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/forms/volunteer_engineer_submission_2015.rb [hunk 1]
```diff
@@ -9,8 +9,9 @@ class VolunteerEngineerSubmission2015 < VolunteerEngineerSubmission
   # and after each Hour of Code.
   UNSUBSCRIBE_HOC = "untilhoc".freeze
   UNSUBSCRIBE_FOREVER = "forever".freeze
-  DEFAULT_DISTANCE = 24 # kilometers
-  DEFAULT_NUM_VOLUNTEERS = 10
+  DEFAULT_DISTANCE_MILES = 20
+  DEFAULT_DISTANCE_KM = 32
```
Comment (breville) [inline pegasus/forms/volunteer_engineer_submission_2015.rb:13]: How are MILES vs KM used in this system?

**Review by islemaster — APPROVED** [review]
I like this approach a lot.

File: pegasus/forms/volunteer_engineer_submission_2015.rb (hunk 1 above)
Comment (hacodeorg) [reply pegasus/forms/volunteer_engineer_submission_2015.rb:13]: We use KM value in SQL query, which uses [ST_Distance_Sphere](https://dev.mysql.com/doc/refman/5.7/en/spatial-convenience-functions.html). 
https://github.com/code-dot-org/code-dot-org/blob/bed88da02d544205a802c941b0665319b9005156/pegasus/forms/volunteer_engineer_submission_2015.rb#L187
We use MILES value in UI.

**Review by breville — APPROVED** [review]

**Review by agealy — APPROVED** [review]

**Comment by wjordan** [comment]
Some UX-related thoughts:

- Add an XHR to https://code.org/v2/client-location and use `zip_code` to get value to set default map location
  - might need to add `coordinates` to the `client-location` response to pass that value instead of `zip_code`
- Use the [Browser Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation) ([available](https://caniuse.com/#feat=geolocation) on all of our supported browsers) to get coordinates to set default map location
  - this might need an extra button/icon for the user to click because browser-permissions are usually required for this

**Comment by hacodeorg** [comment]
Thanks Will for the suggestion. I created a task to track improvement work https://codedotorg.atlassian.net/browse/PLC-658

**Pull request merged**